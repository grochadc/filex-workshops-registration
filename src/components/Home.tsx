import React from "react";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import Alert from "react-bootstrap/Alert";
import { useFormik } from "formik";
import * as Yup from "yup";
import { useMediaQuery } from "react-responsive";

const Home: React.FC<any> = ({ setCurrentRoute, setCode }) => {
  const isTabletOrMobile = useMediaQuery({ query: "(max-width: 400px)" });
  const formik = useFormik({
    initialValues: {
      code: "",
    },
    validationSchema: Yup.object({
      code: Yup.number().required("Required"),
    }),
    onSubmit: (values) => {
      setCode(values.code);
      setCurrentRoute("selection");
    },
  });
  return (
    <div>
      <Form onSubmit={formik.handleSubmit} inline={isTabletOrMobile}>
        <Form.Group>
          <Form.Label htmlFor="code">Code:</Form.Label>
          <Form.Control
            id="code"
            type="text"
            onChange={formik.handleChange}
            value={formik.values.code}
          />
          <Button type="submit">Submit</Button>
        </Form.Group>
      </Form>
      {formik.touched.code && formik.errors.code ? (
        <Alert variant="warning">
          {/code must be a `number` type/.test(formik.errors.code) ? (
            <>Please provide your code as a number</>
          ) : (
            formik.errors.code
          )}
        </Alert>
      ) : null}
    </div>
  );
};

export default Home;
