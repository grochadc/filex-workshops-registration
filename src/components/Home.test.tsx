import React, { Fragment } from "react";
import { render } from "@testing-library/react";
import Home from "./Home";

test("Renders Home without crashing", () => {
  const { asFragment } = render(<Home />);
  expect(asFragment()).toMatchSnapshot();
});
