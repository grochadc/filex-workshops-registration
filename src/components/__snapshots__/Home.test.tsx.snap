// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Home without crashing 1`] = `
<DocumentFragment>
  <div>
    <form
      class=""
    >
      <div
        class="form-group"
      >
        <label
          class="form-label"
          for="code"
        >
          Code:
        </label>
        <input
          class="form-control"
          id="code"
          type="text"
          value=""
        />
        <button
          class="btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
